### PR TITLE
Stop assuming blocking sockets in PyOpenSSL mode

### PR DESCRIPTION
**Pull request by pquentin**

### CODE REVIEW

**Comment by pquentin** [comment]
Well, wrap_socket still assumes blocking mode. I don't know how to fix that one properly yet.

**Comment by njsmith** [comment]
That's OK, we set the socket back to blocking mode when calling `wrap_socket` anyway, because non-blocking handshakes are annoying to deal with, and unnecessary in our case :-)

File: urllib3/_backends/sync_backend.py [hunk 1]
```diff
@@ -69,9 +70,9 @@ def receive_some(self):
         while True:
             try:
                 return self._sock.recv(BUFSIZE)
-            except ssl.SSLWantReadError:
+            except (ssl.SSLWantReadError, OpenSSL.SSL.WantReadError):
                 self._wait(readable=True, writable=False)
-            except ssl.SSLWantWriteError:
+            except (ssl.SSLWantWriteError, OpenSSL.SSL.WantWriteError):
```
[the remark this answers is not in the document]
Comment (pquentin) [reply urllib3/_backends/sync_backend.py]: Oh, yes, that's much better this way: OpenSSL should only be imported in pyopenssl.py.

**Comment by pquentin** [comment]
I addressed the changes, please take a look!

**Pull request merged**